pkg/functions/client.go

```diff
@@ -737,6 +760,30 @@ func (c *Client) Deploy(ctx context.Context, f Function, opts ...DeployOption) (
 		return f, err
 	}
 
+	// If Redeployment to NEW namespace was successful -- undeploy dangling Function in old namespace.
+	// On forced namespace change (using --namespace flag)
+	if f.Namespace != "" && f.Namespace != f.Deploy.Namespace && f.Deploy.Namespace != "" {
+		// TODO: when new prompt is implemented, add here a "are you sure?" check possibly
+		if c.verbose {
+			fmt.Fprintf(os.Stderr, "Info: Deleting old func in '%s' because the namespace has changed to '%s'\n", f.Deploy.Namespace, f.Namespace)
+		}
+
+		// c.Remove removes a Function in f.Deploy.Namespace which removes the OLD Function
+		// because its not updated yet (see few lines below)
+		err = c.Remove(ctx, f, true)
+
+		// Warn when service is not found and set err to nil to continue. Function's
+		// service mightve been manually deleted prior to the subsequent deploy or the
+		// namespace is already deleted therefore there is nothing to delete
+		if ErrFunctionNotFound != err {
```

Comment:
IMHO this is going to really help users trust the code; little warnings like this.